tests/data_context/test_data_context_deprecation.py

```diff
@@ -0,0 +1,158 @@
+import pytest
```

Comment:
this is a fine unit test; what would we need to do to capture the troubles from before in the E2E suite?